src/ddci.c

```diff
@@ -707,9 +707,10 @@ int ddci_create_pmt(ddci_device_t *d, SPMT *pmt, uint8_t *new_pmt, int pmt_size,
 
     // Add Stream pids
     // Add CA IDs and CA Pids
+    adapter *ddci_adapter = get_adapter(d->id);
+    
     for (i = 0; i < pmt->stream_pids; i++) {
-
-        if (get_ca_multiple_pmt(pmt->adapter)) {
+        if (ddci_adapter && get_ca_multiple_pmt(ddci_adapter->id)) {
```

Comment:
is this not always true ?
Every DDCI adapter has a regular adapter attached to it.

Reply:
Yes it is, but Coverity complained. I can drop the commit if there's a way to mark the issue as a non-issue in Coverity?